gateway/sds_gateway/users/views.py

```diff
@@ -1609,6 +1659,173 @@ def _serve_file_download(self, zip_uuid: str, user) -> HttpResponse:
 user_temporary_zip_download_view = TemporaryZipDownloadView.as_view()
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+class UploadFilesView(View):
+    def _handle_file_uploads(self, request, files, relative_paths):
+        saved_files = []
+        errors = []
+        skipped_files = []
+        metadata_only_files = []
+
+        for f, rel_path in zip(files, relative_paths, strict=True):
+            if "/" in rel_path:
+                directory, filename = rel_path.rsplit("/", 1)
+                directory = (
+                    "/" + directory if not directory.startswith("/") else directory
+                )
+            else:
+                directory = "/"
+                filename = rel_path
```

Comment:
use [`pathlib`](https://docs.python.org/3/library/pathlib.html#methods-and-properties) methods and properties: it's less error-prone.